src/iq.rs

```diff
@@ -266,7 +265,7 @@ impl IQClient {
             }
             if result.is_err() {
                 let res_err = result.unwrap_err();
-                if res_err.is_client_error() {
+                if res_err.status().unwrap().is_client_error() {
                     match res_err.status().unwrap() {
```

Comment:
We unwrap this twice. I'd love to reduce that to zero times, but for now should this be?
```suggestion
                let status = res_err.status().unwrap();
                if status.is_client_error() {
                    match status {
```